go.mod

```diff
@@ -20,6 +19,7 @@ require (
 	github.com/mitchellh/mapstructure v1.5.0
 	github.com/nozzle/throttler v0.0.0-20180817012639-2ea982251481
 	github.com/olekukonko/tablewriter v0.0.5
+	github.com/pandatix/go-cvss v0.5.1
```

Comment:
I'm not sure if we should switch to an even less used package than `github.com/goark/go-cvss` here. Thoughts on that @kubernetes/release-managers?

Reply:
I am skeptical as well about moving to a package which is less widely used. Wondering what is the maintenance guarantee for the package.

Reply:
The maintenance of this dependency is part of my job, for both OSS and industrial needs. This imply that if anything is required by the community, I'm ready to do it (as long as it is relative to CVSS).
Also notice that maintenance may not happen as long as CVSS standards (v2, v3.0 and v3.1) are emitted for their own versions and revisions are not on their ways. CVSS v4 will have its own submodule as soon as it is officially published by the first.org CVSS SIG :wink:

Reply:
I would echo the concerns from @saschagrunert and @palnabarun.

@pandatix is there a way you could measure/quantify the ultimate time that we are spending today? If required, I would welcome a PR that logs time spent on CVSS - first - for a true comparison of value vs. risk.

Looking at the 2 code bases, it appears that `pandatix/go-cvss` is 2x the size of `goark/go-cvss` (200kb vs. 100kb) which is a minimal total, but a surprising increase in comparison. 

Is there anything preventing you from PRing your improvements to `goark/go-cvss` as a "best of both worlds"?

/cc @jrsapi

Reply:
TL;DR:
 - binary size **decreased**
 - performances are **better** but does not change so much
 - my implementation is **safer**

Details available [here](https://github.com/kubernetes/release/files/10513768/details.md), so it avoids filling the page with explanations.

---

Conclusion:

Given your questions, yes it would benefit `krel` to use my implementation, but the binary sizes and performances metrics are not so relevant given the experiment conditions.
Nevertheless, what I showed is how goark/go-cvss lacks testing, compliancy validation and hardening.
I understand that a _new_ implementation with a near amount of stars could refrein you from accepting this PR, but it showed performing better both on efficiency and quality/compliancy. When using an implementation to validate inputs, it is expected that it validates only what is indeed valid: it was not the case and could have remained such for years if my own did not detect this.

Making PRs would fix goark/go-cvss until reaching the same code quality and outputs, but even there, reaching the performances will need a complex internal code refactoring... This big work would be easily and fastly fixed with this PR, as it would not be relevant to do the job I did twice.

This change may also impact the sig-security as they would be able to use the same efficient and safer implementation, which is what I also look forward with this PR.

Reply:
My only concern is that one person only maintains this project, and we might want to do a code scanning if we adopt this proposed project, it has the same project name as the current one, and it is not a fork.

The question here is how critical this dependency is for our tool.

Reply:
Another perspective: We don't care about performance here. 

The only place we parse CVSS is when ingesting the CVE data from the SRC advisories which is an extremely rare and manual event. So we need a better reason to increase the bus factor risk in our tooling than performance and marginal binary size.